moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/parallel_gripper_command_controller_handle.hpp

```diff
@@ -0,0 +1,214 @@
+        cmd_state.name.push_back(trajectory.joint_trajectory.joint_names[i]);
+      }
+    }
+
+    if (gripper_joint_indices.empty())
+    {
+      RCLCPP_ERROR(logger_, "Received trajectory does not include joint names that %s can command.", name_.c_str());
+      return false;
+    }
+
+    // send last point
+    int tpoint = trajectory.joint_trajectory.points.size() - 1;
+    RCLCPP_DEBUG(logger_, "Sending command from trajectory point %d.", tpoint);
+
+    // fill in goal from last point
+    for (std::size_t idx : gripper_joint_indices)
+    {
+      if (trajectory.joint_trajectory.points[tpoint].positions.size() <= idx)
+      {
+        RCLCPP_ERROR(logger_, "ParallelGripperCommand expects a joint trajectory with one \
+                               point that specifies at least the position of joint \
+                               '%s', but insufficient positions provided.",
+                     trajectory.joint_trajectory.joint_names[idx].c_str());
+        return false;
+      }
+      cmd_state.position.push_back(trajectory.joint_trajectory.points[tpoint].positions[idx]);
+      // only set the velocity or effort if the user has specified a positive non-zero max value
+      if (max_velocity_ > 0.0)
+      {
+        cmd_state.velocity.push_back(max_velocity_);
+      }
+      if (max_effort_ > 0.0)
+      {
+        cmd_state.effort.push_back(max_effort_);
+      }
+    }
+    rclcpp_action::Client<control_msgs::action::ParallelGripperCommand>::SendGoalOptions send_goal_options;
+    // Active callback
+    send_goal_options.goal_response_callback =
+        [this](const rclcpp_action::Client<control_msgs::action::ParallelGripperCommand>::GoalHandle::SharedPtr&
+               /* unused-arg */) { RCLCPP_DEBUG_STREAM(logger_, name_ << " started execution."); };
+    // Send goal
+    auto current_goal_future = controller_action_client_->async_send_goal(goal, send_goal_options);
+    current_goal_ = current_goal_future.get();
+    if (!current_goal_)
+    {
+      RCLCPP_ERROR(logger_, "%s goal was rejected by server.", name_.c_str());
+      return false;
+    }
+
+    done_ = false;
+    last_exec_ = moveit_controller_manager::ExecutionStatus::RUNNING;
+    return true;
+  }
+
+  void setCommandJoint(const std::string& name)
+  {
+    command_joints_.clear();
+    addCommandJoint(name);
+  }
+
+  void addCommandJoint(const std::string& name)
+  {
+    command_joints_.insert(name);
+  }
+
+  void allowFailure(bool allow)
+  {
+    allow_stalling_ = allow;
+  }
+
+private:
+  void controllerDoneCallback(
+      const rclcpp_action::ClientGoalHandle<control_msgs::action::ParallelGripperCommand>::WrappedResult& wrapped_result)
+      override
+  {
+    if (wrapped_result.code == rclcpp_action::ResultCode::ABORTED && wrapped_result.result->stalled && allow_stalling_)
+    {
+      finishControllerExecution(rclcpp_action::ResultCode::SUCCEEDED);
+    }
+    else
+    {
+      finishControllerExecution(wrapped_result.code);
+    }
```

Comment:
If we end up deciding to remove allow_stalling

```suggestion
      return finishControllerExecution(wrapped_result.code);
```